vault-bin: init at 1.1.3

PsyanticY opened the pull request:
Currently the vault module do not include the ui part of it due to some complications (check this pr: #49082) I think instead of doing the packaging from source we can use the binary Hashicorp is providing that includes the UI.
This is still just a draft and some more work will be made (how to handle darwin and udpate the module with a UI option) if we agree on proceeding using this method. 
 cc @LnL7  @rushmorem  @offline @pradeepchhetri
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

offlinehacker commented:
I do not agree to package vault from prebuilt binary by default, if you have a good reason we can make separate package called `vault-bin`, but I want to first hear which issues you had building vault with UI?

PsyanticY commented:
Hi @offlinehacker  Please have a look here #49082

PsyanticY commented:
After discussion with @lnl7 Proceeding  with a separate.

offlinehacker commented:
I assume it's statically compiled and that's why you didn't need patchelf?

offlinehacker commented on pkgs/tools/security/vault/vault-bin.nix:
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
+  name = "vault-${version}";
```
Can you please change name to `vault-bin-${version}`, if you query profie packages bi name, it will be easier to see which vault have you installed.

offlinehacker commented on pkgs/tools/security/vault/vault-bin.nix:
```diff
@@ -0,0 +1,28 @@
+
+  buildInputs = [ unzip ];
+
+  buildCommand = ''
```
It's better to define installPhase instead. Unzip should be done automatically.

PsyanticY replied:
I tried using `InstallPhase` alone but  nix-build complain abount not knowing how to unpack the archive. Adding `unpackPhase` fixes it. replacing `buildCommand` only add more steps to the build so i thing keeping it is better .

offlinehacker replied:
I think if you add unzip to nativeBuildInputs it should work, take a look here for an example: https://github.com/NixOS/nixpkgs/blob/c126510809c158477690118048bbde7cd83c174b/pkgs/development/interpreters/dart/default.nix#L52

PsyanticY replied:
works, Thanks

offlinehacker commented on pkgs/tools/security/vault/vault-bin.nix:
```diff
@@ -0,0 +1,28 @@
+  meta = with stdenv.lib; {
+    homepage = https://www.vaultproject.io;
+    description = "A tool for managing secrets, this binary includes the UI";
+    platforms = platforms.linux;
```
Please fix platform to only include x86_64-linux, since that's the only package currently being downloaded, or make a conditional fetch based on platform.

PsyanticY replied:
I would like some direction on how to do that (conditional fetch)  since i couldn't really find any examples in nixpkgs

PsyanticY committed:
vault-bin: init at 1.1.3

This binray contain the UI part of HashiCorp Vault that we were not able to build it due to the need to generate a very big yarn file. #49082

PsyanticY commented:
@offlinehacker  yeah it is statically compiled so no need for `patchelf`

offlinehacker commented on pkgs/tools/security/vault/vault-bin.nix:
```diff
@@ -0,0 +1,28 @@
+    description = "A tool for managing secrets, this binary includes the UI";
+    platforms = [ "x86_64-linux" ];
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ rushmorem lnl7 offline pradeepchhetri psyanticy ];
```
Can you please remove other people from maintainers, as I don't know if they want to be included. Leave yourself and you can leave me.

PsyanticY replied:
Done

PsyanticY commented:
@offlinehacker Should be good now Thanks for the help.

offlinehacker commented:
Thanks, looks good now. I will just do a minor stylistic change, and merge to master 🙂

PsyanticY commented:
@offlinehacker  thanks

offlinehacker committed:
vault-bin: use sourceRoot instead of custom unpackPhase
